src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/PeriodicSyncStrategy.java

```diff
@@ -82,6 +137,79 @@ public void clearSyncQueue() {
      */
     @Override
     public int getRemainingCapacity() {
-        return 0;
+        return syncQueue.remainingCapacity();
+    }
+
+    /**
+     * Take and execute items from the sync queue. This is intended to be run in a separate thread.
+     * This will work on all the sync requests in the queue until it's empty.
+     */
+    @SuppressWarnings({"PMD.CompareObjectsWithEquals", "PMD.AvoidCatchingGenericException"})
+    private void syncLoop() {
```

Comment:
is this method any different from the RealTime? 
Seems like the only difference is that it calls `poll` instead of `take`

Could it go in the base class where you give it a `Supplier<SyncRequest>`?

Reply:
Apart from that difference, In the real time sync, the while loop exits when the thread is interrupted. In periodic sync, it exists when the sync queue is empty. So I think we can keep it as is for now.

Reply:
why does the periodic sync not exit when the thread is interrupted?

Does that get it into an infinite loop? The next time it the syncQueue is polled, won't that throw an InterruptedException because the current thread is interrupted? The reentrant lock in the syncQueue calls `lockInterruptibly` which will throw

Reply:
Good point. I will update the code to exit if the thread was interrupted.

Reply:
I think I gave you bad advice - the original `poll` should not have thrown an exception since it was using the regular lock (not lockInterruptibly). I must have looked at the other implemented method when checking before.

`poll` is intended to not block so you don't need to worry about it interrupting, whereas `poll(timeout)` does block.

If poll doesn't throw an interrupted exception, then the request will either be null or the thread will be interrupted and exit